src/backend/InvenTree/order/serializers.py

```diff
@@ -325,6 +336,99 @@ def save(self):
         order.complete_order()
 
 
+class PurchaseOrderRequestApprovalSerializer(serializers.Serializer):
+    """Serializer for requesting approval of a purchase order by eligible users."""
+
+    class Meta:
+        """Metaclass options."""
+
+        fields = []
+
+    def save(self):
+        """Save the serializer to request approval of the order."""
+        order = self.context['order']
+        request = self.context['request']
+        order.request_approval(request.user)
+
+
+class PurchaseOrderStatePermissionsSerializer(serializers.Serializer):
+    """Serializer indicating what permissions the user has for limited-permission states."""
+
+    class Meta:
+        """Metaclass options."""
+
+        model = order.models.PurchaseOrder
+        fields = ['can_approve', 'can_issue']
+
+    def get_can_approve(self, instance):
+        """Run model method to check approval permission."""
+        request = self.context['request']
+
+        return instance.approval_allowed(request.user)
+
+    def get_can_issue(self, instance):
+        """Run model method to check issue permission."""
+        request = self.context['request']
+
+        return instance.allowed_to_issue(request.user)
+
+    can_approve = serializers.SerializerMethodField()
+
+    can_issue = serializers.SerializerMethodField()
+
+
+class PurchaseOrderReadySerializer(serializers.Serializer):
+    """Serializer to mark a purchase order as ready to issue."""
+
+    class Meta:
+        """Metaclass options."""
+
+        fields = []
+
+    def save(self):
+        """Save the serializer mark the order as ready."""
+        order = self.context['order']
+        request = self.context['request']
+        order.mark_order_ready(request.user)
+
+
+class PurchaseOrderRejectSerializer(serializers.Serializer):
+    """Serializer for rejecting a purchase order."""
+
+    class Meta:
+        """Metaclass options."""
+
+        fields = ['reject_reason']
+
+    reject_reason = serializers.CharField(
+        label=_('Reason for rejection'),
+        help_text=_('Message outlining the reason for rejecting the order'),
+        default='',
+        required=False,
+        allow_blank=True,
+    )
+
+    def save(self):
+        """Save the serializer to 'reject' the order."""
+        order = self.context['order']
+        reason = self.data['reject_reason']
```

Comment:
What happens here if 'reject_reason' is not supplied? Do you need to use `self.data.get('reject_reason', '')` instead?

Reply:
I haven't tried not supplying it, actually. It defaults to the value of the field in both CUI and PUI. (empty string if no reason is typed in)
I'll attempt some pure API calls and see what happens